src/main/java/edu/ucsb/cs56/ucsb_courses_search/service/UCSBFinalService.java

```diff
@@ -0,0 +1,120 @@
+package edu.ucsb.cs56.ucsb_courses_search.service;
+
+import java.util.Arrays;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Service object that wraps the UCSB Academic Curriculum Finals API
+ */
+@Service
+public class UCSBFinalService implements FinalService {
+
+    private Logger logger = LoggerFactory.getLogger(UCSBFinalService.class);
+
+    private String apiKey;
+
+    public UCSBFinalService(@Value("${ucsb.api.consumer_key}") String apiKey) {
+        this.apiKey = apiKey;
+        logger.info("apiKey=" + apiKey);
+    }
+
+    public String setQuarter(String quarter){
```

Comment:
setQuarter is probably not the best name, because it makes the user think that this is setting some internet state or variable. You may want to rename this to something more descriptive.